source/extensions/filters/listener/original_dst/config.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "envoy/extensions/filters/listener/original_dst/v3/original_dst.pb.h"
+#include "envoy/extensions/filters/listener/original_dst/v3/original_dst.pb.validate.h"
+#include "envoy/network/filter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ListenerFilters {
+namespace OriginalDst {
+
+class Config {
+public:
+  Config() = default;
+  Config(const envoy::extensions::filters::listener::original_dst::v3::OriginalDst& config,
+         const envoy::config::core::v3::TrafficDirection& traffic_direction)
+      : method_(getOriginalDstMethod(config)), traffic_direction_(traffic_direction) {}
+
+  enum class OriginalDstMethod {
+    SocketOption = 0,
+    NoOp = 1,
+  };
+  OriginalDstMethod method_{OriginalDstMethod::SocketOption};
+  envoy::config::core::v3::TrafficDirection traffic_direction_;
```

Comment:
Could you make these data members private or change the type of this object from `class` to `struct`?

Is it possible to mark them `const`?